packages/docusaurus/src/server/plugins/init.ts

```diff
@@ -44,39 +45,127 @@ export default function initPlugins({
 
   const plugins: InitPlugin[] = pluginConfigs
     .map((pluginItem) => {
-      let pluginModuleImport: string | undefined;
+      // let pluginModuleImport: string | undefined;
       let pluginOptions: PluginOptions = {};
+      // let plugin: PluginModule | undefined;
+      let validateOptions;
+      // let pluginModule: ImportedPluginModule | undefined;
+      let pluginVersion: DocusaurusPluginVersionInformation;
 
       if (!pluginItem) {
         return null;
       }
 
-      if (typeof pluginItem === 'string') {
-        pluginModuleImport = pluginItem;
-      } else if (Array.isArray(pluginItem)) {
-        [pluginModuleImport, pluginOptions = {}] = pluginItem;
+      function getPlugin(): PluginModule | null {
+        if (typeof pluginItem === 'function') {
+          return pluginItem;
+        } else if (
```

Comment:
I think it would be simpler to handle all array cases under a single if statement